packages/trpc/server/routers/viewer/teams/setInviteExpiration.handler.ts

```diff
@@ -29,13 +29,15 @@ export const setInviteExpirationHandler = async ({ ctx, input }: SetInviteExpira
     throw new TRPCError({ code: "UNAUTHORIZED" });
 
   const oneDay = 24 * 60 * 60 * 1000;
-  const expires = expiresInDays ? new Date(Date.now() + expiresInDays * oneDay) : new Date();
+  const expires = expiresInDays
+    ? new Date(Date.now() + expiresInDays * oneDay)
+    : new Date("9999-12-31T23:59:59Z"); //maximum possible date incase the link is set to never expire
```

Comment:
can you explain why this works and why the previous doesnt?

Reply:
Looks like it’s because when expiresInDays is false, it was setting the link expiration date to now. i.e. it immediately expired

Reply:
https://github.com/calcom/cal.com/blob/95e037c664d119b8a5d2cdec99bd33264e8f9797/packages/features/ee/teams/components/InviteLinkSettingsModal.tsx#L47-L54

For never expire link, expiresInDays is undefined

> Looks like it’s because when expiresInDays is false, it was setting the link expiration date to now. i.e. it immediately expired

Then this happens